### PR TITLE
refactor: add schemas to user admin controller

### PR DESCRIPTION
**Pull request by olav**

### CODE REVIEW

File: src/lib/services/user-service.ts [hunk 1]
```diff
@@ -171,7 +171,7 @@ class UserService {
         return { ...user, rootRole: roleId };
     }
 
-    async search(query: IUserSearch): Promise<IUser[]> {
+    async search(query: string): Promise<IUser[]> {
```
Comment (ivarconr) [inline src/lib/services/user-service.ts:174]: Nice catch!

File: src/lib/services/user-splash-service.ts [hunk 2]
```diff
@@ -20,24 +20,22 @@ export default class UserSplashService {
         this.logger = getLogger('services/user-splash-service.js');
     }
 
-    async getAllUserSplashs(user: User): Promise<Object> {
+    async getAllUserSplashes(user: User): Promise<Record<string, boolean>> {
```
Comment (ivarconr) [inline src/lib/services/user-splash-service.ts:23]: nice!

File: src/lib/openapi/spec/me-schema.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { FromSchema } from 'json-schema-to-ts';
+import { userSchema } from './user-schema';
+import { permissionSchema } from './permission-schema';
+import { feedbackSchema } from './feedback-schema';
+
+export const meSchema = {
+    $id: '#/components/schemas/meSchema',
+    type: 'object',
+    additionalProperties: false,
+    required: ['user', 'permissions', 'feedback', 'splash'],
```
Comment (ivarconr) [inline src/lib/openapi/spec/me-schema.ts:10]: Q: is it intuitive that "array" types are required? I assume they can be an empty list.

File: src/lib/routes/admin-api/user-admin.ts [hunk 4]
```diff
@@ -113,118 +264,127 @@
...
-                q && q.length > 1 ? await this.userService.search(q) : [];
-
-            if (this.anonymise) {
-                users = this.anonymiseUsers(users);
-            }
-            res.json(users);
-        } catch (error) {
-            this.logger.error(error);
-            res.status(500).send({ msg: 'server errors' });
+        let users = q && q.length > 1 ? await this.userService.search(q) : [];
```
[the remark this answers is not in the document]
Comment (ivarconr) [reply src/lib/routes/admin-api/user-admin.ts]: Not related to this PR, but would be good to resolve it regardless. Up for that @olav ?

[the remark this answers is not in the document]
Comment (olav) [reply src/lib/routes/admin-api/user-admin.ts]: Definitely! Should be fixed by https://github.com/Unleash/unleash/pull/1692/commits/fc577b1dae30ff934e4c0e8ed3030e8e5ee6111c.

File: src/lib/openapi/spec/create-user-schema.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { FromSchema } from 'json-schema-to-ts';
+
+export const createUserSchema = {
+    $id: '#/components/schemas/createUserSchema',
+    type: 'object',
+    additionalProperties: false,
+    required: ['rootRole'],
```
Comment (ivarconr) [inline src/lib/openapi/spec/create-user-schema.ts:7]: we actually also require `username` or `email`. But that might not be possible to express in OAS?

Comment (ivarconr) [reply src/lib/openapi/spec/create-user-schema.ts:7]: tbh, I think we should require email. That's also what we do require in the UI at the moment.

**Review by ivarconr — APPROVED** [review]

**Pull request merged**